src/bindgen/ir/ty.rs

```diff
@@ -330,13 +337,17 @@ impl Type {
 
     pub fn simplify_option_to_ptr(&mut self) {
         let mut simplified = None;
-
-        if let &mut Type::Path(ref mut path) = self {
-            if path.name == "Option" && path.generics.len() == 1 && path.generics[0].is_repr_ptr() {
-                simplified = Some(path.generics.pop().unwrap());
+        if let &mut Type::Path(ref generic) = self {
+            let option_path = Path::new("Option");
+            if generic.path() == &option_path
+                && generic.generics().len() == 1
+                && generic.generics()[0].is_repr_ptr()
```

Comment:
This isn't your fault, but the logic of this function is wrong. It will convert `Option<* const i32>` to `const int32_t*` which isn't correct. The transformation is only valid if the pointer is non-nullable (like &i32). I'll file an issue.

Reply:
Good to know. I'll keep it as is then.